### ISSUE TITLE
Different behavior of secured API keys between 2.x and 3.x

### ISSUE THREAD

**Issue by tampix** [algolia/algoliasearch-client-java#657]
- Algolia Client Version: 3.4.0
- Language Version: 11.0.4
- Instantsearch.js: 1.11.15

### Description

While migrating from 2.16.8 to 3.4.0, I encountered a strange behavior with secured api keys, and can't tell if the problem lies with us (ie. Java code is buggy, Instantsearch.js is too old...) or if it's a regression in the 3.x.

#### With 2.16.8

```java
final Query query = new Query()
    .setTagFilters(Arrays.asList("(87,1033,1052,1534)"))
    .setUserToken("70");
return client.generateSecuredApiKey(searchApiKey, query);
```

The decoded generated key is: `<MAC>tagFilters=%2887%2C1033%2C1052%2C1534%29&userToken=70`

Once used with Instantsearch.js, the query has the intended behavior and result contains:
```json
{
  "results": [
    {
      "query": "",
      "params": "query=&hitsPerPage=20&page=0&getRankingInfo=1&&tagFilters=&tagFilters=%2887%2C1033%2C1052%2C1534%29&userToken=70",
      "parsedQuery": ""
    }
  ]
}
```

#### With 3.4.0

```java
final Query query = new Query()
    .setTagFilters(Arrays.asList(Arrays.asList("87", "1033", "1052", "1534")))
    .setUserToken("70");
final SecuredApiKeyRestriction restriction = new SecuredApiKeyRestriction()
    .setQuery(query);
return client.generateSecuredAPIKey(searchApiKey, restriction);
```

The decoded generated key is: `<MAC>query=userToken%3D70%26tagFilters%3D%255B%255B%252287%2522%252C%25221033%2522%252C%25221052%2522%252C%25221534%2522%255D%255D`

Once used with Instantsearch.js, the query doesn't have the intended behavior and result contains:
```json
{
  "results": [
    {
      "query": "userToken=70&tagFilters=%5B%5B%2287%22%2C%221033%22%2C%221052%22%2C%221534%22%5D%5D",
      "params": "query=&hitsPerPage=20&page=0&getRankingInfo=1&tagFilters=&query=userToken%3D70%26tagFilters%3D%255B%255B%252287%2522%252C%25221033%2522%252C%25221052%2522%252C%25221534%2522%255D%255D",
      "parsedQuery": "usertoken 70 tagfilters 5b 5b 2287 22 2c 221033 22 2c 221052 22 2c 221534 22 5d 5d"
    }
  ]
}
```

I notice a few things that could be problematic:

* The empty `&tagFilters=` before the `&query` might be the source of the problem, which would mean the problem lies in the client's code (the one using Instantsearch.js)
* The `parsedQuery` seems strange
* `tagFilters` in the `query` is url encoded twice

**Comment by Ant-hem** [comment]
Hi @tampix 

Thanks for detailed issue. 

I investigated on my side and there is indeed an issue the way the query string is generated. I'll work on a fix referencing this issue and I'll publish a patch version right after.

Keep an eye on the issue!

**Comment by Ant-hem** [comment]
Fixed with `3.6.1`. Feel free to update, there is no BC.
